utils/self-extracting-executable/compressor.cpp

```diff
@@ -411,8 +482,16 @@ int main(int argc, char* argv[])
     }
     ++start_of_files;
 
-    if (copy_decompressor(argv[0], output_fd))
-        return 1;
+    if (decompressor != nullptr && decompressor[0] != 0)
```

Comment:
Looking at l. 459, I think that at this point `decompressor[0]` is never `0` if `decompressor` is `!= nullptr`. I.e., the second condition in the if statement is always true.